### PR TITLE
Add (back) telepath over TLS

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -296,27 +291,21 @@ def getCellIden(self):
...
-        # with SSL a valid client cert sets ssl:user
-        name = link.get('ssl:user')
-        if name is not None:
-            return self.auth.users.get(name)
-
-        # fall back on user/passwd
         auth = mesg[1].get('auth')
         if auth is None:
-            return None
+            raise s_exc.AuthDeny(mesg='Unable to find cell user')
```
Comment (vEpiphyte) [inline synapse/lib/cell.py]: `mesg='User not present in link.'`

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -296,27 +291,21 @@ def getCellIden(self):
...
 
         name, info = auth
 
         user = self.auth.users.get(name)
         if user is None:
-            return None
+            raise s_exc.AuthDeny(mesg='Unable to find cell user')
```
Comment (vEpiphyte) [inline synapse/lib/cell.py]: add `, user=user`

File: synapse/lib/certdir.py [hunk 3]
```diff
@@ -825,6 +831,44 @@ def signUserCsr(self, xcsr, signas, outp=None):
         name = xcsr.get_subject().CN
         return self.genUserCert(name, csr=pkey, signas=signas, outp=outp)
 
+    def _loadCasIntoSSLContext(self, ctx):
+        path = s_common.genpath(self.certdir, 'cas')
+        for name in os.listdir(path):
+            if name.endswith('.crt'):
+                ctx.load_verify_locations(os.path.join(path, name))
+
+    def getClientSSLContext(self) -> ssl.SSLContext:
```
Comment (vEpiphyte) [inline synapse/lib/certdir.py]: type hinting syntax - agreed upon?

File: synapse/lib/certdir.py [hunk 4]
```diff
@@ -825,6 +831,44 @@ def signUserCsr(self, xcsr, signas, outp=None):
...
+        '''
+        Returns an ssl.SSLContext appropriate for initiating a TLS session
+        '''
+        sslctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+        self._loadCasIntoSSLContext(sslctx)
+
+        return sslctx
+
+    def getServerSSLContext(self, hostname: Optional[str] = None) -> ssl.SSLContext:
```
Comment (vEpiphyte) [inline synapse/lib/certdir.py]: type hinting syntax - agreed upon?

File: synapse/tests/utils.py [hunk 5]
```diff
@@ -641,6 +602,10 @@ def thisHostMustNot(self, **props):  # pragma: no cover
     async def getTestDmon(self, mirror='dmontest'):
 
         with self.getTestDir(mirror=mirror) as dirn:
+
+            # Copy test certs
+            shutil.copytree(os.path.join(self.getTestFilePath(), 'certdir'), os.path.join(dirn, 'certs'))
```
Comment (vEpiphyte) [inline synapse/tests/utils.py]: `shutil.copytree(self.getTestFilePath('certdir'), os.path.join(dirn, 'certs'))`

should work fine

File: synapse/lib/cell.py (hunk 1 above)
Comment (jnwatson) [reply synapse/lib/cell.py]: ok

File: synapse/tests/utils.py (hunk 5 above)
Comment (jnwatson) [reply synapse/tests/utils.py]: ok

**Comment by jnwatson** [comment]
Added an ssl cert failure in test_telepath.

File: synapse/lib/cell.py (hunk 2 above)
Comment (jnwatson) [reply synapse/lib/cell.py]: `user=name` (user is None)

**Pull request merged**